### PR TITLE
Add API reference section

### PR DESCRIPTION
**Pull request by krzysztofwolski**
PR is quite big, so probably it would be easier to review commit by commit instead going thru all changed files at once.

image
image
image

### CODE REVIEW

File: docs/api-reference/directives/deprecated.mdx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+<span
+style={{
+      backgroundColor: color,
+      borderRadius: '2px',
+      color: '#fff',
+      padding: '0.2rem',
+    }}>
+{children}
+</span>
+);
```
Comment (patrys) [inline docs/api-reference/directives/deprecated.mdx:16]: What's that bit? It seems to be duplicated for each individual page, can we replace this with an import? Should this use theme colors?

Comment (krzysztofwolski) [reply docs/api-reference/directives/deprecated.mdx:16]: image
Thats the bit responsible for rendering deprecated badge. Will migrate it to common stylesheet 👍

Comment (krzysztofwolski) [reply docs/api-reference/directives/deprecated.mdx]: @patrys fixed

**Comment by patrys** [comment]
Task linked: [SALEOR-2155 Generate API Reference for docs](https://app.clickup.com/t/2549495/SALEOR-2155)

File: website/docusaurus2-graphql-doc-generator/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+npm install --save @edno/docusaurus2-graphql-doc-generator
+```
+
+### Yarn
+
+```shell
+yarn add @edno/docusaurus2-graphql-doc-generator
+```
+
+Then you add it in your site's `docusaurus.config.js`'s plugins option:
```
Comment (karolkielecki) [inline website/docusaurus2-graphql-doc-generator/README.md:25]: Then you need to add it to the plugin options in `docusaurus.config.js` of your site.

File: website/docusaurus2-graphql-doc-generator/README.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        schema: "https://swapi.graph.cool/",
+        rootPath: "./docs", // docs will be generated under './docs/swapi' (rootPath/baseURL)
+        baseURL: "swapi",
+        homepage: "./docs/swapi.md",
+      },
+    ],
+};
+```
+
+Each option is described in the section [Options](#options).
```
Comment (karolkielecki) [inline website/docusaurus2-graphql-doc-generator/README.md:56]: All the options are described in the [Options](#options) section.

Comment (karolkielecki) [inline website/docusaurus2-graphql-doc-generator/README.md:56]:
```suggestion
All the options are described in the [Options](#options) section.
```

File: website/docusaurus2-graphql-doc-generator/README.md (hunk 2 above)
Comment (karolkielecki) [inline website/docusaurus2-graphql-doc-generator/README.md:25]: Then you need to add it to the plugin options in  `docusaurus.config.js` of your site.

Comment (karolkielecki) [inline website/docusaurus2-graphql-doc-generator/README.md:25]:
```suggestion
Then you need to add it to the plugin options in  `docusaurus.config.js` of your site:
```

File: docs/api-reference/api-introduction.mdx [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+title: API Reference
+sidebar_label: Introduction
+---
+
+View complete documentation of our GraphQL API.
```
Comment (patrys) [inline docs/api-reference/api-introduction.mdx:6]: I think this file is unused?

Comment (krzysztofwolski) [reply docs/api-reference/api-introduction.mdx:6]: Removed reference landing page from main directory (docs/api-reference), modified script and now its place is in "docs/api-reference/api-introduction.mdx". Also file contains now date of generation.

File: README.md [hunk 5]
```diff
@@ -188,6 +188,15 @@ For more information about the navigation bar, click [here](https://docusaurus.i
 
 For more information about custom pages, click [here](https://docusaurus.io/docs/en/custom-pages).
 
+# Updating Api Reference
```
Comment (patrys) [inline README.md:191]:
```suggestion
# Updating the API Reference
```

File: README.md [hunk 6]
```diff
@@ -188,6 +188,15 @@ For more information about the navigation bar, click [here](https://docusaurus.i
...
+
+All files in /docs/api-reference are generated by @edno/docusaurus2-graphql-doc-generator script. Introduction page is automatically copied from `/docs/api-introduction.mdx` file.
+
+To update API reference directory:
```
Comment (patrys) [inline README.md:195]:
```suggestion
To update the API reference:
```

File: docs/index.mdx [hunk 7]
```diff
@@ -22,3 +22,11 @@ Here you’ll find everything you need to start working with [Saleor](https://sa
 [![](icons/getting-started.svg) **Developer's Guide** Build and deploy a store based on Saleor.](developer.mdx)
 
 </CardGrid>
+
+## Api Reference
+
+<CardGrid home>
+
+[![](icons/getting-started.svg) **Api Reference** Detailed description of the whole schema.](api-reference/api-introduction.mdx)
```
Comment (patrys) [inline docs/index.mdx:30]:
```suggestion
[![](icons/getting-started.svg) **API Reference** A detailed description of the entire GraphQL API.](api-reference/api-introduction.mdx)
```

Also, there is a better icon that was removed when the previous reference was dropped: https://github.com/mirumee/saleor-docs/commit/962b221e4ca5935ee6a5fbe511f35e18f7774062

**Review by patrys — CHANGES_REQUESTED** [review]

File: docs/index.mdx [hunk 8]
```diff
@@ -22,3 +22,11 @@ Here you’ll find everything you need to start working with [Saleor](https://sa
 [![](icons/getting-started.svg) **Developer's Guide** Build and deploy a store based on Saleor.](developer.mdx)
 
 </CardGrid>
+
+## Api Reference
```
[the remark this answers is not in the document]
Comment (krzysztofwolski) [reply docs/index.mdx]: Moved Api reference into dev docs

**Review by patrys — APPROVED** [review]

**Pull request merged**